src/cockpit/packages.py

```diff
@@ -317,11 +319,13 @@ def get_content_security_policy(self, origin):
     def serve_file(self, path, channel):
         if path == 'po.js':
             # We do locale-dependent lookup only for /po.js.  This always succeeds.
+            # Keep the file in memory for efficient lookups and manifests.js processing
             locales = parse_accept_language(channel.headers)
             data, (content_type, encoding) = find_translation(self.translations, locales)
         else:
             # Otherwise, just look up the file based on its path.  May raise KeyError.
-            data, (content_type, encoding) = self.files[path]
+            path, (content_type, encoding) = self.files[path]
+            data = path.read_bytes()
```

Comment:
Please make sure you also handle `OSError` from the same place that `KeyError` is handled in `Packages.serve_package_file()`.  It might even make sense to return different messages depending on what is wrong.

Reply:
Done.